llpc/context/llpcCompiler.cpp

```diff
@@ -1936,30 +1936,24 @@ Result Compiler::BuildRayTracingPipeline(const RayTracingPipelineBuildInfo *pipe
     // Raytracing modules: pipeline shader count + entryModule
     unsigned modulesCount = pipelineInfo->shaderCount + 1;
 
-    // Set pipeline has trace ray module by default, as long as the pipeline contains any shader.
-    pipelineOut->hasTraceRay = (pipelineInfo->shaderCount > 0);
-    unsigned i = 0;
-
-    if (pipelineInfo->rtState.pipelineFlags & VK_PIPELINE_CREATE_LIBRARY_BIT_KHR) {
-      // Library pipeline usually does not have trace ray module
-      pipelineOut->hasTraceRay = false;
-      for (i = 0; i < pipelineInfo->shaderCount; ++i) {
-        const auto &shaderInfo = pipelineInfo->pShaders[i];
-        // Check raygen spirv module have OpTraceRayKHR or not
-        if (shaderInfo.entryStage == ShaderStageRayTracingRayGen) {
-          const ShaderModuleData *moduleData = reinterpret_cast<const ShaderModuleData *>(shaderInfo.pModuleData);
-          if (moduleData->usage.hasTraceRay) {
-            pipelineOut->hasTraceRay = true;
-            break;
-          }
-        }
+    bool hasRayQuery = false;
+    pipelineOut->hasTraceRay = false;
+    for (unsigned i = 0; i < pipelineInfo->shaderCount; ++i) {
+      const auto &shaderInfo = pipelineInfo->pShaders[i];
+      const ShaderModuleData *moduleData = reinterpret_cast<const ShaderModuleData *>(shaderInfo.pModuleData);
+      if (moduleData->usage.hasTraceRay) {
+        pipelineOut->hasTraceRay = true;
+        break;
```

Comment:
This early exit may leave `hasRayQuery` with the wrong value, which does not matter currently as we only depend on `hasRayQuery` if `pipelineOut->hasTraceRay == false`, but could matter in the future if further usages are added?

Maybe rename `hasRayQuery` to reflect that, or replace it by `hasRayQueryOrTraceRay`?